rdsbroker/config.go

```diff
@@ -29,6 +30,10 @@ func (c Config) Validate() error {
 		return errors.New("Must provide a non-empty BrokerName")
 	}
 
+	if c.AWSPartition == "" {
+		c.AWSPartition = "aws"
```

Comment:
There are two issues with this assignment here: 
- Validate method is not a best place to set defaults
- `c Config` is not passed by reference but a value so that assignment is local copy which makes [RDSDBInstance.partition](https://github.com/jmcarp/paas-rds-broker/blob/0dd6bb7771d6f1ca6190d4a070ce453628d1ba38/awsrds/rds_db_instance.go#L17) empty. That later fails [here](https://github.com/jmcarp/paas-rds-broker/blob/0dd6bb7771d6f1ca6190d4a070ce453628d1ba38/awsrds/rds_db_instance.go#L90) because [dbInstanceARN](https://github.com/jmcarp/paas-rds-broker/blob/0dd6bb7771d6f1ca6190d4a070ce453628d1ba38/awsrds/rds_db_instance.go#L459) returns wrong `arn`